olp-cpp-sdk-dataservice-read/tests/StreamLayerClientImplTest.cpp

```diff
@@ -281,6 +296,319 @@ TEST(StreamLayerClientImplTest, SubrcribeCancelOnClientDestroy) {
   }
 }
 
+TEST(StreamLayerClientImplTest, Unsubsribe) {
+  auto network_mock = std::make_shared<NetworkMock>();
+  auto cache_mock = std::make_shared<CacheMock>();
+  olp::client::OlpClientSettings settings;
+  settings.network_request_handler = network_mock;
+  settings.cache = cache_mock;
+
+  {
+    SCOPED_TRACE("Unsubsribe success");
+
+    StreamLayerClientImpl client(kHRN, kLayerId, settings);
+
+    {
+      SetupNetworkExpectation(*network_mock, kUrlLookupStream,
+                              kHttpResponseLookupStream,
+                              olp::http::HttpStatusCode::OK);
+
+      SetupNetworkExpectation(
+          *network_mock, kUrlStreamSubscribe, kHttpResponseSubscribe,
+          olp::http::HttpStatusCode::CREATED, RequestMethod::POST);
+
+      std::promise<SubscribeResponse> promise;
+      std::future<SubscribeResponse> future(promise.get_future());
+      client.Subscribe(SubscribeRequest(), [&](SubscribeResponse response) {
+        promise.set_value(response);
+      });
+
+      ASSERT_EQ(future.wait_for(kTimeout), std::future_status::ready);
+
+      const auto& response = future.get();
+      ASSERT_TRUE(response.IsSuccessful());
+
+      ASSERT_EQ(response.GetResult(), kSubcriptionId);
+    }
+    {
+      SetupNetworkExpectation(*network_mock, kUrlLookupStream,
+                              kHttpResponseLookupStream,
+                              olp::http::HttpStatusCode::OK);
+
+      SetupNetworkExpectation(*network_mock, kUrlStreamUnsubscribe,
+                              kHttpResponseEmpty, olp::http::HttpStatusCode::OK,
+                              RequestMethod::DELETE);
+
+      std::promise<UnsubscribeResponse> promise;
+      std::future<UnsubscribeResponse> future(promise.get_future());
+      client.Unsubscribe(
+          [&](UnsubscribeResponse response) { promise.set_value(response); });
+
+      ASSERT_EQ(future.wait_for(kTimeout), std::future_status::ready);
+
+      const auto& response = future.get();
+      ASSERT_TRUE(response.IsSuccessful());
+
+      ASSERT_EQ(response.GetResult(), kSubcriptionId);
+    }
+
+    Mock::VerifyAndClearExpectations(network_mock.get());
+  }
+
+  {
+    SCOPED_TRACE("Unsubsribe fails since the second call");
```

Comment:
What does "... since the second call" means?
Please change and try to be more specific.

Reply:
I completely over-thought this case. The idea was to check something after the unsubscribe done. I removed the test case for now. Maybe I will add something similar later on (like calling "poll" after unsubscribe)